core/src/main/java/org/elasticsearch/search/aggregations/metrics/scripted/ScriptedMetricAggregatorBuilder.java

```diff
@@ -47,6 +63,38 @@ public ScriptedMetricAggregatorBuilder(String name) {
     }
 
     /**
+     * Read from a stream.
+     */
+    public ScriptedMetricAggregatorBuilder(StreamInput in) throws IOException {
+        super(in, InternalScriptedMetric.TYPE);
+        initScript = in.readOptionalStreamable(Script.SUPPLIER);
+        mapScript = in.readOptionalStreamable(Script.SUPPLIER);
+        combineScript = in.readOptionalStreamable(Script.SUPPLIER);
+        reduceScript = in.readOptionalStreamable(Script.SUPPLIER);
```

Comment:
Maybe in another PR but can we rename `Script.SUPPLIER` to something else (maybe `Script.STREAM_READER` or  `Script.READER`) since `SUPPLIER doesn't really tell me what it is for?